src/components/Common/UI/FormField/FormField.tsx

```diff
@@ -0,0 +1,69 @@
+import { useTranslation } from 'react-i18next'
+import classNames from 'classnames'
+import styles from './FormField.module.scss'
+import { createMarkup } from '@/helpers/formattedStrings'
+
+export type BaseFormFieldProps = {
+  description?: React.ReactNode
+  errorMessage?: string
+  isRequired?: boolean
+} & (
+  | {
+      label?: string
+      'aria-label'?: never
+    }
+  | {
+      'aria-label': string
+      label?: never
+    }
+)
+
+type FormFieldProps = {
+  children: React.ReactNode
+  htmlFor: string
+  errorMessageId: string
+  className?: string
+} & BaseFormFieldProps
+
+export const FormField: React.FC<FormFieldProps> = ({
+  label,
+  description,
+  errorMessage,
+  errorMessageId,
+  children,
+  isRequired = false,
+  htmlFor,
+  className,
+}: FormFieldProps) => {
+  const { t } = useTranslation('common')
+
+  return (
```

Comment:
Yeah I think this makes sense as a rendering component the way you set it up so far.